src/reducers/coordinateReducer/coordinateActions.ts

```diff
@@ -0,0 +1,7 @@
+const COORDINATE_ACTIONS = {
+  added: "added",
+  updated: "updated",
+  removed: "removed",
+} as const;
```

Comment:
Would you provide the enum way of defining the type?

Reply:
I would use [const assertions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) rather than enum when I need to use the type and variable in general but enum is also a good way to do the same thing. Because I think const assertion has more type safety and flexibility to usage, also has better compatibility in some cases. According to the typescript official document then also mentions that between [Objects and Enums](https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums) and they said may not need an enum when an object with as const could suffice. That's why I use the const assertions.

I have a tech sharing about the enum and const assertions in typescript last month. If you have any thoughts please feel free to let me know.
https://hackmd.io/@jacklee/B1M-Dc50q#/

Reply:
Oh great! That's sensible to me.